cuda_core/cuda/core/experimental/_module.py

```diff
@@ -106,50 +106,61 @@ class ObjectCode:
 
     """
 
-    __slots__ = ("_handle", "_code_type", "_module", "_loader", "_sym_map")
+    __slots__ = ("_handle", "_backend_version", "_jit_options", "_code_type", "_module", "_loader", "_sym_map")
     _supported_code_type = ("cubin", "ptx", "ltoir", "fatbin")
 
     def __init__(self, module, code_type, jit_options=None, *, symbol_mapping=None):
         if code_type not in self._supported_code_type:
             raise ValueError
         _lazy_init()
+
+        # handle is assigned during _lazy_load
         self._handle = None
+        self._jit_options = jit_options
+
+        self._backend_version = "new" if (_py_major_ver >= 12 and _driver_ver >= 12000) else "old"
+        self._loader = _backend[self._backend_version]
+
+        self._code_type = code_type
+        self._module = module
+        self._sym_map = {} if symbol_mapping is None else symbol_mapping
 
-        backend = "new" if (_py_major_ver >= 12 and _driver_ver >= 12000) else "old"
-        self._loader = _backend[backend]
+    # TODO: do we want to unload in a finalizer? Probably not..
 
-        if isinstance(module, str):
+    def _lazy_load_module(self, *args, **kwargs):
+        if self._handle is not None:
+            return
+        if isinstance(self._module, str):
```

Comment:
@ksimpson-work I have a nit here: Starting this line could you alias the used attributes like this
```python
jit_options = self._ jit_options
module = self._module
if isinstance(_module, str):
    ...
```
so that we can 
1. keep the diff minimal
2. avoid excessive attribute access